src/ATen/native/xpu/UnaryOps.cpp

```diff
@@ -1028,11 +1028,12 @@ Tensor& XPUNativeFunctions::ceil_(Tensor& self) {
 }
 
 Tensor& XPUNativeFunctions::ceil_out(const Tensor& self, Tensor& out) {
+  auto iter = ceil_meta(self, out);
```

Comment:
I am very curious about what happens if we don't reorder these two lines of code. And why this reorder can affect the result?

Reply:
The test case expects to catch a runtime error such as "RuntimeError: Found dtype Float but expected Long", which is raised in ceil_meta. According to the original code logic, if the input type is int, ceil_meta will not be called.

Reply:
Make sense. Thanks very much~